disneylandClient/worker.py

```diff
@@ -93,6 +92,12 @@ def run(self):
 
                 self.acquire_cpus(process_name, 1)
 
-                p = Process(name=process_name, target=self.do_job, args=(job,))
+                p = Process(name=process_name, target=self.do_job, args=(job, queue))
                 self.processes.append(p)
                 p.start()
+
+            for p in self.processes:
+                p.join()
```

Comment:
This line will block for until child process terminates. Lets say you take first `p` in `self.processes` and wait for it. All other processes might already be finished, but you will block on this line. Moreover, this block will prevent worker from pulling new jobs.